src/wrappers/google/pubsub/SubscriptionExecutor.ts

```diff
@@ -0,0 +1,112 @@
+    WrapperUtils.finishInvocationData(execContext, pluginContext);
+}
+
+/**
+ * Start trace
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function startTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+
+    const {
+        tracer,
+        request,
+    } = execContext;
+
+    const contextInformation: any = execContext.getContextInformation();
+
+    const propagatedSpanContext: ThundraSpanContext =
+        tracer.extract(opentracing.FORMAT_TEXT_MAP, request.attributes) as ThundraSpanContext;
+
+    const triggerOperationName = get(request, `attributes.${TriggerHeaderTags.RESOURCE_NAME}`)
+        || contextInformation.operationName;
+    const traceId = get(propagatedSpanContext, 'traceId') || Utils.generateId();
+    const incomingSpanID = get(propagatedSpanContext, 'spanId');
+
+    const rootSpan = tracer._startSpan(triggerOperationName, {
+        propagated: propagatedSpanContext ? true : false,
+        parentContext: propagatedSpanContext,
+        rootTraceId: traceId,
+        domainName: contextInformation.applicationDomainName,
+        className: contextInformation.applicationClassName,
+    });
+
+    rootSpan.isRootSpan = true;
+
+    const subscription = request._subscriber._subscription;
+    const subscriptionName = (subscription.metadata && subscription.metadata.topic)
+        ? subscription.metadata.topic : subscription.name;
+
+    const maskMessage = ConfigProvider.get<boolean>(ConfigNames.THUNDRA_TRACE_INTEGRATIONS_GOOGLE_PUBLISH_BODY_MASK);
+    const tags = {
+      [GooglePubSubTags.PROJECT_ID]: subscription.pubsub ? subscription.pubsub.projectId : '',
+      [GooglePubSubTags.SUBSCRIPTION]: subscriptionName,
+      ...( !maskMessage ? { [GooglePubSubTags.MESSAGE]:  GooglePubSubUtils.parseMessage(request) } : undefined ),
+      [SpanTags.SPAN_TYPE]: SpanTypes.GOOGLEPUBSUB,
+    };
+
+    rootSpan.addTags(tags);
+
+    InvocationSupport.setAgentTag(SpanTags.TRIGGER_OPERATION_NAMES, [triggerOperationName]);
+    InvocationSupport.setAgentTag(SpanTags.TRIGGER_DOMAIN_NAME, DomainNames.API);
+    InvocationSupport.setAgentTag(SpanTags.TRIGGER_CLASS_NAME, ClassNames.GOOGLEPUBSUB);
+
+    if (incomingSpanID) {
+        InvocationTraceSupport.addIncomingTraceLink(incomingSpanID);
+    }
+
+    execContext.traceId = traceId;
+    execContext.rootSpan = rootSpan;
+    execContext.spanId = execContext.rootSpan.spanContext.spanId;
+    execContext.rootSpan.startTime = execContext.startTimestamp;
+    execContext.triggerOperationName = triggerOperationName;
+}
+
+/**
+ * Finish trace
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function finishTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+    WrapperUtils.finishTrace(pluginContext, execContext);
+
+    const { rootSpan, request } = execContext;
+    rootSpan.tags['pubsub.ack'] = request._handled ? 1 : 0;
```

Comment:
- `google.pubsub.ack` 
- value should be `true` or `false`
- better to define it in the constants